packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -268,38 +337,97 @@ const service = (dependencies: BaseDependencies) => {
 		};
 	};
 
+	/**
+	 * Fetches submitted data from the database based on the provided category ID, organization, pagination options, and optional filter options.
+	 *
+	 * This function retrieves a list of submitted data associated with the specified `categoryId` and `organization`.
+	 * It supports a view representation, pagination and optional filtering using a structured query (`sqon`) or entity names.
+	 * The result includes both the fetched data and metadata such as the total number of records and an error message if applicable.
+	 *
+	 * @param categoryId - The ID of the category for which data is being fetched.
+	 * @param organization - The name of the organization to filter the data by.
+	 * @param paginationOptions - An object containing pagination options, such as page number and items per page.
+	 * @param filterOptions - Optional filtering options.
+	 * @param filterOptions.sqon - An optional Structured Query Object Notation (SQON) for advanced filtering criteria.
+	 * @param filterOptions.entityName - An optional array of entity names to filter the data by. Can include undefined entries.
+	 * @param filterOptions.view - An optional flag indicating the view type
+	 * @returns A promise that resolves to an object containing:
+	 * - `data`: An array of `SubmittedDataResponse` objects, representing the fetched data.
+	 * - `metadata`: An object containing metadata about the fetched data, including the total number of records.
+	 *   If an error occurs during data retrieval, `metadata` will include an `errorMessage` property.
+	 */
 	const getSubmittedDataByOrganization = async (
 		categoryId: number,
 		organization: string,
 		paginationOptions: PaginationOptions,
-		filterOptions?: { sqon?: SQON; entityName?: (string | undefined)[] },
+		filterOptions: { sqon?: SQON; entityName?: (string | undefined)[]; view: ViewType },
```

Comment:
`view` here is also optional

Also there's enough overlap between `getSubmittedDataByCategory, getSubmittedDataByOrganization & getSubmittedDataBySystemId` that `filterOptions` could be defined as a reusable variable